### PR TITLE
Fix quoted links giving blank page on click

### PR DESCRIPTION
**Pull request by riking**
https://meta.discourse.org/t/links-in-quotes-sometimes-not-working/24540/

### CODE REVIEW

File: app/controllers/clicks_controller.rb [hunk 1]
```diff
@@ -5,21 +5,28 @@ class ClicksController < ApplicationController
...
+        params.merge!({ user_id: current_user.id }) if current_user.present?
         @redirect_url = TopicLinkClick.create_from(params)
+
+        if @redirect_url.blank? && params[:url].index('?')
+          # Check the url without query parameters
+          params[:url].sub!(/\?.*$/, '')
+          @redirect_url = TopicLinkClick.create_from(params)
+        end
       end
+    rescue => e
```
Comment (SamSaffron) [inline app/controllers/clicks_controller.rb]: hmmm, why is this needed?

**Comment by SamSaffron** [comment]
can I see some examples in the wild of the problem this is solving (make a meta post)

**Comment by coding-horror** [comment]
It's this

https://meta.discourse.org/t/links-in-quotes-sometimes-not-working/24540

File: app/controllers/clicks_controller.rb (hunk 1 above)
Comment (riking) [inline app/controllers/clicks_controller.rb]: Removed

File: db/migrate/20150129082106_add_from_quote_to_topic_link.rb [hunk 2]
```diff
@@ -0,0 +1,5 @@
+class AddFromQuoteToTopicLink < ActiveRecord::Migration
+  def change
+    add_column :topic_links, :from_quote, :boolean, default: false, null: false
```
Comment (SamSaffron) [inline db/migrate/20150129082106_add_from_quote_to_topic_link.rb]: I dunno, this is adding a feature we don't really need to add just to fix a bug, can we just fix bug instead?

Comment (riking) [inline db/migrate/20150129082106_add_from_quote_to_topic_link.rb]: There's another problem hiding here - yt-light creates links in javascript that don't show up in the cooked, so they also give the "link not found" white screen.

I now agree, this is the wrong way to fix it.

**Comment by SamSaffron** [comment]
I am closing this for now, let's come up with another fix
